src/components/Trading/TradingForm.tsx

```diff
@@ -160,22 +131,99 @@ function TradingForm(props: Props) {
     ? BigNumber(secondaryBalance.balance).sub(secondaryBalance.asset_type === "native" ? minAccountBalance : 0)
     : BigNumber(0)
 
-  const assets = React.useMemo(() => props.trustlines.map(balancelineToAsset), [props.trustlines])
-
   const maxPrimaryAmount =
-    props.primaryAction === "buy"
+    primaryAction === "buy"
       ? spendableSecondaryBalance.gt(0) && effectivePrice.gt(0)
         ? BigNumber(spendableSecondaryBalance).div(effectivePrice)
         : BigNumber(0)
       : spendablePrimaryBalance.gt(0)
       ? BigNumber(spendablePrimaryBalance)
       : BigNumber(0)
 
-  const isDisabled =
-    !primaryAsset || primaryAmount.lte(0) || sellingAmount.gt(sellingBalance.balance) || effectivePrice.lte(0)
+  return {
+    defaultPrice,
+    effectivePrice,
+    maxPrimaryAmount,
+    minAccountBalance,
+    primaryAmount,
+    primaryBalance,
+    relativeSpread,
+    secondaryAmount,
+    secondaryBalance,
+    spendablePrimaryBalance,
+    spendableSecondaryBalance
+  }
+}
+
+interface TradingFormValues {
+  primaryAsset: Asset | undefined
+  primaryAmountString: string
+  secondaryAsset: Asset
+  manualPrice: string
+}
+
+interface Props {
+  account: Account
+  accountData: AccountData
+  className?: string
+  dialogActionsRef: RefStateObject
+  initialPrimaryAsset?: Asset
+  primaryAction: "buy" | "sell"
+  sendTransaction: (transaction: Transaction) => void
+  style?: React.CSSProperties
+  trustlines: Horizon.BalanceLineAsset[]
+}
+
+function TradingForm(props: Props) {
+  const classes = useStyles()
+  const isSmallScreen = useIsMobile()
+  const isSmallHeightScreen = useMediaQuery("(max-height: 500px)")
+  const isSmallScreenXY = isSmallScreen || isSmallHeightScreen
+
+  const [expanded, setExpanded] = React.useState(false)
+  const [priceMode, setPriceMode] = React.useState<"primary" | "secondary">("secondary")
+
+  const { control, errors, formState, getValues, handleSubmit, register, setValue, triggerValidation, watch } = useForm<
```

Comment:
[Minor] Just a quick thought: Since the returned object is quite complex and we need a lot of its props it might be worth trying a slightly different style of using it.

Let's try to not destructure it: `const form = useForm<TradingFormValues>(/*...*/)` and then later access things as `form.errors`, etc.